### PR TITLE
feat: Add Mailchain Email Provider #4336

### PR DESCRIPTION
**Pull request by ItIsOHM**
### What change does this PR introduce?

This PR adds Mailchain provider to Novu by using the Mailchain SDK. #4336 

I was able to add the Mailchain Provider to the Integration Store, but there a few bugs which need fixing so I'm keeping this PR as draft for now.

Any help is appreciated as i'm a beginner here.

Screenshots of the Integration Store:
image
image

After connecting to the mailchain sdk using the secret key provided by mailchain:
image

### CODE REVIEW

**Comment by davidsoderberg** [comment]
Please provide some screenshots of how it looks on integration page.

**Comment by ItIsOHM** [comment]
> Please provide some screenshots of how it looks on integration page.

Okay i'll run the local env again and add the screenshots here.

**Comment by davidsoderberg** [comment]
Ah sorry if I was unclear but the fallback change should not be committed only changed locally so you can take screenshots with the new integrations view.

**Comment by ItIsOHM** [comment]
> Ah sorry if I was unclear but the fallback change should not be committed only changed locally so you can take screenshots with the new integrations view.

Ahh I see! That's okay I'll revert that commit and then test locally ASAP. Also, I'm getting an "Invalid Mnemonic" error while using the Mailchain provider (i was trying to send a test mail). Can you please look into that if possible?

File: packages/application-generic/package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
     "@novu/gupshup": "^0.20.0-alpha.0",
     "@novu/infobip": "^0.20.0-alpha.0",
     "@novu/kannel": "^0.20.0-alpha.0",
+    "@novu/mailchain" : "0.16.3",
```
Comment (Cliftonz) [inline packages/application-generic/package.json:65]: This needs to be the same version as all of the other providers

File: providers/mailchain/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Novu Mailchain Provider
+
+A Mailchain email provider library for [@novu/node](https://github.com/novuhq/novu)
+
+## Usage
+
+```javascript
+    FILL IN THE INITIALIZATION USAGE
```
Comment (Cliftonz) [inline providers/mailchain/README.md:8]: This usage should be filled in

File: providers/mailchain/package.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
+{
+  "name": "@novu/mailchain",
+  "version": "0.16.3",
```
Comment (Cliftonz) [inline providers/mailchain/package.json:3]: This version should be the same as the rest of the providers

**Review by Cliftonz — CHANGES_REQUESTED** [review]
Good work, fix the few things I mentioned and we should be good to go.

File: packages/application-generic/package.json (hunk 1 above)
Comment (ItIsOHM) [reply packages/application-generic/package.json]: Why is every version the same, if I may ask?

**Comment by ItIsOHM** [comment]
Hey there @Cliftonz, I have pushed the files with the necessary changes, and added the square logos too for the new integration view. Although I'm not able to see the square logos when running locally. Any idea why @davidsoderberg?

**Comment by ItIsOHM** [comment]
Hello there, what should my next step be? I'm not able to test Mailchain (or any other provider, for that matter) as I'm getting something like a "provider issue" when trying to send a test mail.

**Comment by Cliftonz** [comment]
@ItIsOHM  Can you post the error you are getting?

**Comment by ItIsOHM** [comment]
> @ItIsOHM Can you post the error you are getting?

Hey there, this is the error I'm getting in development locally:
image
It is the same for Resend too. I couldn't test the others.

I checked the network logs, and i'm getting a 400 error:
image

This is the preview:
image

And this is the payload:
image

I will further look in the `test-email.e2e.ts` file to see if I can understand the root of this issue.

**Comment by ItIsOHM** [comment]
Hey @Cliftonz I'm really sorry, but I am unable to find why the API isn't working correctly on my end. I looked into the API directory to see if I can spot any errors, but I couldn't find anything which may cause this issue. I'd love if you can take a look in this and help me out.

Also very sorry for the delay, I'll respond as soon as possible from now on 😅

**Comment by Cliftonz** [comment]
If you fix the conflicts and make sure it is up to date, I am sure someone else can test this.

**Comment by ItIsOHM** [comment]
Hey i'm currently resolving these new conflicts, sorry for the mess.

**Comment by Cliftonz** [comment]
@ItIsOHM looks like cspell needs updated

**Comment by ItIsOHM** [comment]
> @ItIsOHM looks like cspell needs updated

Hey i've changed the cspell version here to the one used in all other packages, should I also update the `~` to `^` or is it alright?

**Comment by Cliftonz** [comment]
> > @ItIsOHM looks like cspell needs updated
> 
> Hey i've changed the cspell version here to the one used in all other packages, should I also update the `~` to `^` or is it alright? image

My bad, I ment that the cspell file needs to be updated to pass the ci/cd pipeline here.
https://github.com/novuhq/novu/actions/runs/6616690791/job/17971466071?pr=4427

**Comment by ItIsOHM** [comment]
Ahh okay, so I basically need to add `mailchain` and `Mailchain` to the cspell file in `/.cspell.json`, like this?
image

**Comment by Cliftonz** [comment]
> Ahh okay, so I basically need to add `mailchain` and `Mailchain` to the cspell file in `/.cspell.json`, like this? image

Correct, however its better to add them to the bottom.

**Comment by ItIsOHM** [comment]
Hey @Cliftonz, what should my next step be?

**Comment by Cliftonz** [comment]
@ItIsOHM  Looks like first we need to resolve the conflicts, once you do that @ me and we can go to the next step.

**Comment by Cliftonz** [comment]
@ItIsOHM Looks like we are missing a work in the cspell file.
initialise

Ping me when you pushed a fix, apologies for taking so long.

**Comment by ItIsOHM** [comment]
> @ItIsOHM Looks like we are missing a work in the cspell file. initialise
> 
> Ping me when you pushed a fix, apologies for taking so long.

Oh no it's alright, I'll do it ASAP as I'm a bit busy for a few days :)

**Comment by ItIsOHM** [comment]
@Cliftonz I think the test is failing again, what might the reason be?

**Comment by Cliftonz** [comment]
@ItIsOHM  I think the tests are fixed, can you fix the conflicts and we can look at merging this.

**Comment by ItIsOHM** [comment]
Hey @Cliftonz I think the conflicts are resolved now! Please have a look :)

File: .cspell.json [hunk 4]
```diff
@@ -534,6 +536,7 @@
     "tspan",
     "maildata",
     "brevo",
+    "initialise"
```
Comment (rifont) [inline .cspell.json:539]: We are using American English, this shouldn't be required (`initialize`)
```suggestion
```

**Review by rifont — COMMENTED** [review]
Thank you for your contribution!

It appears there are a few unexpected [spell check issues](https://github.com/novuhq/novu/actions/runs/7901350276/job/21564728755?pr=4427) with your PR, alongside [pnpm](https://github.com/novuhq/novu/actions/runs/7901350276/job/21564728500?pr=4427#step:5:198) and [Node version errors](https://github.com/novuhq/novu/actions/runs/7901350276/job/21564728500?pr=4427#step:5:209).

Could you please perform a full install using `pnpm install`, and commit the resulting `pnpm-lock.yaml` changes. Let's see if that resolves the issue.

**Comment by scopsy** [comment]
Closing due to inactivity

**Pull request closed without merging**